### PR TITLE
More updates for Julia 0.7

### PR DESCRIPTION
**Pull request by dpo**
Supersedes #68.

### CODE REVIEW

**Comment by dpo** [comment]
TRON is now failing to converge, so those tests fail.
@abelsiqueira

File: src/solver/tron.jl [hunk 1]
```diff
@@ -309,9 +309,9 @@ function projected_newton!{T <: Real}(x::AbstractVector{T}, H::Union{AbstractMat
     status = stats.status
 
     # Projected line search
-    @views xfree = x[ifree]
+    xfree = x[ifree]
```
Comment (abelsiqueira) [inline src/solver/tron.jl]: This needs to be `@view`.

File: src/solver/tron.jl [hunk 2]
```diff
@@ -309,9 +309,9 @@ function projected_newton!{T <: Real}(x::AbstractVector{T}, H::Union{AbstractMat
...
     @views w = projected_line_search!(xfree, ZHZ, gfree, st, ℓ[ifree], u[ifree])
-    @views s[ifree] += w
+    @views s[ifree] .+= w
```
Comment (abelsiqueira) [inline src/solver/tron.jl]: Also, I think this `@views` isn't required anymore.

**Comment by abelsiqueira** [comment]
Using `@view` fixed a problem. There is some other error that I'm looking into.

File: src/solver/tron.jl (hunk 1 above)
Comment (dpo) [reply src/solver/tron.jl]: `@views` was giving me an error, but I guess `@views` and `@view` are two different things?!

Comment (abelsiqueira) [reply src/solver/tron.jl]: There's possibly some change on 0.7. Using `xfree = @view x[ifree]` works.

**Comment by abelsiqueira** [comment]
Here is the fix for the other errors: https://github.com/JuliaSmoothOptimizers/Optimize.jl/pull/68/commits/85242ee9657cd889806f820e1b514cd3f11040c7
I've pushed to #68 to test.

**Comment by abelsiqueira** [comment]
The error is that (Preallocated) LinearOperators and ForwardDiff are not working together, because the memory used by the linear operator can't store Dual numbers.

**Comment by dpo** [comment]
Thank you. Does that mean we have to use operators without preallocation when using AD models?

**Comment by abelsiqueira** [comment]
We can't have the objective function or constraints using LinearOperators. e.g. `f(x) = dot(x, A * x)` and `c(x) = A * x - b` only work if `A` is a matrix or non-allocating linear operators. Its a `ForwardDiff` limitation.
However, we can have operators with preallocation after the differentiation, such as `hess_op!` and `jac_op`. Those aren't broken if `f` is accepted by `ForwardDiff`.

**Comment by dpo** [comment]
```
ERROR: The following package names could not be resolved:
 * Krylov (not found in project, manifest or registry)
Please specify by known `name=uuid`.
```
The new `Pkg3` is really frustrating.

**Comment by abelsiqueira** [comment]
Looks like we'll need a Krylov release.

**Pull request closed without merging**